examples/cloud_discovery_service/library_api/c/cloud_discovery_service_library_api.c

```diff
@@ -0,0 +1,129 @@
+/*******************************************************************************
+ (c) 2023 Copyright, Real-Time Innovations, Inc. All rights reserved.
+ RTI grants Licensee a license to use, modify, compile, and create derivative
+ works of the Software. Licensee has the right to distribute object form only
+ for use with RTI products. The Software is provided "as is", with no warranty
+ of any type, including any warranty for fitness for any purpose. RTI is under
+ no obligation to maintain or support the Software. RTI shall not be liable for
+ any incidental or consequential damages arising out of the use or inability to
+ use the software.
+ ******************************************************************************/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "ndds/ndds_c.h"
+#include "clouddiscoveryservice/clouddiscoveryservice_service.h"
+
+static int service_main(
+        const char *cfgName,
+        const char *cfgFile,
+        int runForSecs)
+{
+    struct RTI_CDS_Service *service = NULL;
+    struct RTI_CDS_Property property = RTI_CDS_Property_INITIALIZER;
+    DDS_Boolean ok = DDS_BOOLEAN_FALSE;
+    struct DDS_Duration_t runningPeriod = { 0, 0 };
+
+    runningPeriod.sec = runForSecs;
+
+    property.cfg_file = DDS_String_dup(cfgFile);
+    property.service_name = DDS_String_dup(cfgName);
+    property.application_name =
+            DDS_String_dup("CloudDiscoveryService-LibraryAPI-C");
+
+    /* Dial up the logging at service level */
+    property.service_verbosity = RTI_CDS_LOG_VERBOSITY_ALL;
+    property.dds_verbosity = RTI_CDS_LOG_VERBOSITY_EXCEPTIONS;
+
+
+    service = RTI_CDS_Service_new(&property);
+    if (service == NULL) {
+        printf("Error creating the Cloud Discovery Service instance\n");
+        return EXIT_FAILURE;
+    }
+
+    ok = RTI_CDS_Service_start(service);
+    if (!ok) {
+        printf("Error starting the Cloud Discovery Service instance\n");
+        return EXIT_FAILURE;
+    }
+
+    printf("ConfigName=%s, ConfigFile=%s\n", cfgName, cfgFile);
+
+    /* Sleep for the running period */
+    NDDS_Utility_sleep(&runningPeriod);
+
+    ok = RTI_CDS_Service_stop(service);
+    if (!ok) {
+        printf("Error stopping the Cloud Discovery Service instance\n");
+        return EXIT_FAILURE;
+    }
+
+    /* Deleting the service instance for resource cleanup */
```

Comment:
You won't get here if some of the error conditions return. You can either use a `goto done` or create a cleanup function.

Reply:
I moved this to a `service_shutdown()` function.